testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/KeycloakQuarkusServerDeployableContainer.java

```diff
@@ -338,6 +375,40 @@ private long getStartTimeout() {
         return TimeUnit.SECONDS.toMillis(configuration.getStartupTimeoutInSeconds());
     }
 
+    private void killChildProcessesOnWindows(boolean isForced) {
+        for (ProcessHandle childProcessHandle : container.children().collect(Collectors.toList())) {
```

Comment:
I'm fixing the logic here in the PR you are reviewing that makes it possible to run the dist testsuite on Windows. Basically:

* Iterating over the `descendants` is better. It also includes direct children.
* It should be more reliable to collect all futures and at the end wait for completion. Note that processes are terminated asynchronously and in parallel. Doing like that gives a bit more time to Windows to terminate the process (yer, windows has some nuances)
* Note that `join` and using a `get` with a clear timeout behaves differently. The latter should give more control and not block undefinetly. It also handles thread interruption better. From my PoV, it provided a more stable runtime than using `join`.
* Also note from the PR herein mentioned that we have a `sleep` to make sure Windows can terminate processes properly. Don't ask me why we need that but without it, termination is unstable.

Reply:
lets get #13472 in and afterwards i'll adjust